src/shared/components/QueryBuilder/domain/states/idle.ts

```diff
@@ -14,22 +16,30 @@ export const idle: StateNodeConfig<
   entry: [
     'cancelAllLoad',
     assign({
+      // eslint-disable-next-line sonarjs/cognitive-complexity
       suggestions: (context) => {
         let placeholder = MESSAGES.FILTER_ADD;
+        let allowCustomValue = false;
         const lastFilter = getLastFilter(context.filters);
 
         if (lastFilter && isPartialFilter(lastFilter)) {
           if (!lastFilter.operator) {
             placeholder = MESSAGES.SELECT_OPERATOR;
           } else {
-            placeholder =
-              lastFilter?.operator.value === OperatorKind.in ? MESSAGES.SELECT_VALUES : MESSAGES.SELECT_VALUE;
+            allowCustomValue = isRegexOperator(lastFilter.operator.value);
+
+            placeholder = isMultipleValuesOperator(lastFilter.operator.value)
+              ? MESSAGES.SELECT_VALUES
+              : allowCustomValue
+              ? MESSAGES.TYPE_VALUE
+              : MESSAGES.SELECT_VALUE;
           }
         }
 
         return {
           ...defaultContext.suggestions,
           placeholder,
+          allowCustomValue,
```

Comment:
This prevents a `<Select />` component to be rendered when a regex operator has been selected:
<img width="1040" alt="image" src="https://github.com/user-attachments/assets/47c09274-fa88-48c8-9148-44de3b1d46a9">